### PR TITLE
Chunked source separation

### PR DESCRIPTION
**Pull request by Tomiinek**

### CODE REVIEW

File: src/tidytunes/models/external/silerovad.py [hunk 1]
```diff
@@ -103,7 +103,11 @@ def forward(self, audio_chunk_16khz: torch.Tensor, state: list[torch.Tensor]):
 
         x = audio_chunk_16khz.unsqueeze(1)
 
-        x = F.pad(x, (self.audio_padding_size, self.audio_padding_size), mode="reflect")
+        x = F.pad(
+            x.float().contiguous(),
```
Comment (Tomiinek) [inline src/tidytunes/models/external/silerovad.py:107]: Because of https://discuss.pytorch.org/t/reflectionpad2d-input-tensor-must-fit-into-32-bit-index-math/111099/2

File: src/tidytunes/bin/process_audio.py [hunk 2]
```diff
@@ -36,6 +36,9 @@ def process_audio(audios, device, pipeline_components):
 
     for name, func, kwargs, filter_fn in pipeline_components:
 
+        # to improve batching efficiency
+        audio_segments = sorted(audio_segments, key=lambda x: x.duration)
```
Comment (Tomiinek) [inline src/tidytunes/bin/process_audio.py:40]: We cannot sort in `to_batches`, because the filtering and partition functions expect the original order of audios.

File: src/tidytunes/models/source_separation.py [hunk 3]
```diff
@@ -9,104 +10,126 @@ class SourceSeparator(nn.Module):
...
-        Args:
-            audio (B, T): Normalized input audio.
+        audio_buffer = torch.zeros(x.shape[0], self.overlap_samples, device=x.device)
+        window_buffer = None
+        output = []
 
-        Returns:
-            (B, sources, T): Separated sources.
-        """
+        for i in range((x.shape[-1] - self.overlap_samples) // self.segment_samples):
```
Comment (Tomiinek) [inline src/tidytunes/models/source_separation.py:75]: Instead of running demucs on overlaping 10 segments and energy windows on the whole demucsed outputs; we run everything inside the segments so long inputs should not cause OOMs as long as output frames can fit into the memory.

**Comment by Tomiinek** [comment]
> Should this result in the same outputs like before the PR? If yes, maybe we could validate by running on a set of recordings before and after.

No it should not. The source separation should give a bit different outputs, because:
- the segment size and segment overlap has different sizes (a multiple of the energy window frame shift)
- the segment overlap is fixed so that it is like they wanted in the torchaudio comment 
- there is an extra `minimal_energy: float = 1e-7` to fix unstable outputs when there is just a little of energy (the old code just normalizes like `energy / energy.sum()`

Here are energy plots from the first 17 seconds of https://www.youtube.com/watch?v=gKO8f79Ekew

Old:
image

New:
image

Note the missing last energy data point in the old plot! This is because it is NaN due to the division by zero (the window has 0 pad so it is likely NaN at the end of every processed recording)
